frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -2178,3 +2179,12 @@ func validateUsedOnce(c instructions.Command, loc *instructionTracker, warn lint
 	}
 	loc.MarkUsed(c.Location())
 }
+
+func reportInvalidBaseImagePlatform(name string, expected, actual ocispecs.Platform, location []parser.Range, warn linter.LintWarnFunc) {
```

Comment:
Can you rename this something like `validateBaseImagePlatform` or something that implies that it is performing a check? The current naming implies that the check has already happened and this is just reporting the error.